store/v2/commitment/store_test_suite.go

```diff
@@ -166,3 +167,63 @@ func (s *CommitStoreTestSuite) TestStore_Pruning() {
 		}
 	}
 }
+
+func (s *CommitStoreTestSuite) TestStore_Upgrades() {
+	storeKeys := []string{storeKey1, storeKey2, storeKey3}
+	commitDB := dbm.NewMemDB()
+	commitStore, err := s.NewStore(commitDB, storeKeys, log.NewNopLogger())
+	s.Require().NoError(err)
+
+	latestVersion := uint64(10)
+	kvCount := 10
+	for i := uint64(1); i <= latestVersion; i++ {
+		kvPairs := make(map[string]corestore.KVPairs)
+		for _, storeKey := range storeKeys {
+			kvPairs[storeKey] = corestore.KVPairs{}
+			for j := 0; j < kvCount; j++ {
+				key := []byte(fmt.Sprintf("key-%d-%d", i, j))
+				value := []byte(fmt.Sprintf("value-%d-%d", i, j))
+				kvPairs[storeKey] = append(kvPairs[storeKey], corestore.KVPair{Key: key, Value: value})
+			}
+		}
+		s.Require().NoError(commitStore.WriteChangeset(corestore.NewChangesetWithPairs(kvPairs)))
+		_, err = commitStore.Commit(i)
+		s.Require().NoError(err)
+	}
+
+	// create a new commitment store with upgrades
+	upgrades := &corestore.StoreUpgrades{
+		Added: []string{"newStore1", "newStore2"},
+		Renamed: []corestore.StoreRename{
+			{OldKey: storeKey1, NewKey: "renamedStore1"},
+		},
+		Deleted: []string{storeKey3},
+	}
+	newStoreKeys := []string{"store1", "store2", "store3", "renamedStore1", "newStore1", "newStore2"}
+	realStoreKeys := []string{"renamedStore1", "store2", "newStore1", "newStore2"}
+	commitStore, err = s.NewStore(commitDB, newStoreKeys, log.NewNopLogger())
+	s.Require().NoError(err)
+	err = commitStore.LoadVersion(latestVersion, upgrades)
+	s.Require().NoError(err)
+
+	// apply the changeset again
```

Comment:
If I got it correct, you seeded some stores with test data, execute a migration via loadVersion but I don't fully understand what you are doing next? Can you elaborate on this? I was expecting some verification on the store data.